packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -152,17 +253,30 @@ class MethodChannel {
   /// The `T` type argument is the expected return type. It is treated as
   /// nullable.
   @optionalTypeArgs
-  Future<T?> _invokeMethod<T>(String method, { required bool missingOk, dynamic arguments }) async {
+  Future<T?> _invokeMethod<T>(String method,
+      {required bool missingOk, dynamic arguments}) async {
     assert(method != null);
-    final ByteData? result = await binaryMessenger.send(
-      name,
-      codec.encodeMethodCall(MethodCall(method, arguments)),
-    );
+    final ByteData input =
+        codec.encodeMethodCall(MethodCall(method, arguments));
+    assert(() {
+      if (debugProfilePlatformChannels) {
+        _debugRecordUpStream(this, '$name#$method', codec, input);
+      }
+      return true;
+    }());
+    final ByteData? result = await binaryMessenger.send(name, input);
+    assert(() {
+      if (debugProfilePlatformChannels) {
+        _debugRecordDownStream(this, '$name#$method', codec, result);
+      }
+      return true;
+    }());
     if (result == null) {
       if (missingOk) {
         return null;
       }
-      throw MissingPluginException('No implementation found for method $method on channel $name');
+      throw MissingPluginException(
+          'No implementation found for method $method on channel $name');
```

Comment:
here and elsewhere: don't use dartfmt (although it may produce somewhat acceptable results if you do a line length of 120 or something)